ow_materials/include/MaterialDatabase.h

```diff
@@ -0,0 +1,42 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef MATERIAL_DATABASE_H
+#define MATERIAL_DATABASE_H
+
+#include <string>
+#include <ostream>
+#include <unordered_map>
+
+#include <Material.h>
+
+namespace ow_materials
+{
+
+class MaterialDatabase
+{
+public:
+  MaterialDatabase() = default;
+  ~MaterialDatabase() = default;
+
+  MaterialDatabase(const MaterialDatabase&) = delete;
+  MaterialDatabase& operator=(const MaterialDatabase&) = delete;
+
+  void addMaterial(Material mat);
+
+  size_t size();
+
+  inline const Material &getMaterial(MaterialID id) const {
+    return m_database.at(id);
```

Comment:
Assuming you're certain that `id` will be valid, else the `at` would throw an exception.  Minor point: `id` should be `const` per your philosophy though I certainly prefer it as is!

Reply:
I added a comment that notes this may throw an `std::out_of_range`. Presently the method is not called anywhere, but I do intend for it to only ever be called inside of a `try` block.